### ISSUE TITLE
lake representation (for now)?

### ISSUE THREAD

**Issue by mikejohnson51** [mikejohnson51/hyRelease#6]
hi @hellkite,  @lcunha0118,  @dblodgett-usgs

Here is a first stab at a way to represent lakes/reservoirs in an attempt to mirror NWM functionality. It is intended as a first pass, not a final :)

I put ??? in the lake ID for "security". 

Otherwise, the variables are those in the LAKEPARM.nc file, with the exception of `member_wbs`,`partial_areas`, and an updated `LkArea`.

The `member_wbs` are those that have some relation to the lake. The `partial_area` is the area in sqkm of that lake in each catchment where `wb-1` is the flow path for `cat-1`. It seemed redundant to put `member_cats`, but it does require that inference.

The name of the JSON element is the outlet wb-ID.

```json
"wb-12020": {
    "member_wbs": ["wb-11887", "wb-11886", "wb-12020"],
    "partial_areas": [0.3497, 1.1545, 3.3868],
    "lake_id": [?????],
    "Dam_Length": [10],
    "LkMxE": [36.66],
    "OrificeE": [35.9933],
    "WeirE": [36.56],
    "LkArea": [4.891],
    "WeirC": [0.4],
    "WeirL": [10],
    "OrificeC": [0.1],
    "OrificeA": [1],
    "lat": [44.41],
    "lon": [-69.6547],
    "time": [0],
    "ascendingIndex": [3291],
    "ifd": [0.9]
  }
```

Thoughts?

**Comment by dblodgett-usgs** [comment]
This makes sense to me, yes. The nuance here relevant to HY_Features is that _I think_ we are kind of conflating HY_Lake and HY_Reservoir with these attributes. The orifice and weir attributes and those that speak to storage characteristics are logically associated with the function of the lake as a storage (reservoir). 

I do like connecting to the outlet flowpath / waterbody. We are essentially saying "we have a thing that is a lake that impacts the network as a reservoir". It attaches at the identified flowpath and overlays some other set of flowpaths. 

Does that track with the way you are thinking about it?

**Comment by mikejohnson51** [comment]
Yep! Thats great - nicely said. I _believe_ the conflation is only in how the NWM is using LAKE. My understanding is that these are all reservoirs, but the name of the file (LAKEPARAMS) and the assigned ID (lake_id) conflates the two. The ID is really just the waterbodyCOMID from the NHD so we can change the label to "reservoir_id" if it makes this more clear and doesn't mess up t-route?

**Comment by dblodgett-usgs** [comment]
I'm fine with naming being out of sync with conceptual reality ... but not everyone is. ;) 

In HY_Features there is no "lake" feature. Only waterbodies of type "lake". A waterbody of type lake can be the storage part of a reservoir feature. ... so as long as that is how it's done in the NWM for now, let's not sweat the details.

**Comment by mikejohnson51** [comment]
perfect :) thanks @dblodgett-usgs!

**Comment by mikejohnson51** [comment]
Updates from meeting with @hellkite500:

- Update `partial_area` to be `partial_length`
- Move partial_area to catchments, with a `tot_sqkm`, `land_sqkm`, and `wb_sqkm`
- For now, only include Waterbodies in the `LAKEPARAM` file but be able to add all waterbodies latter
- Remove list structure from all but list elements

**Comment by mikejohnson51** [comment]
@hellkite500

```json
"wb-12020": {
    "member_wbs": ["wb-11886", "wb-11887", "wb-12020"],
    "partial_length_km": [1.6882, 0.8351, 3.5525],
    "lake_id":  ?????,
    "Dam_Length": 10,
    "LkMxE": 36.66,
    "OrificeE": 35.9933,
    "WeirE": 36.56,
    "LkArea": 4.5521,
    "WeirC": 0.4,
    "WeirL": 10,
    "OrificeC": 0.1,
    "OrificeA": 1,
    "lat": 44.41,
    "lon": -69.6547,
    "time": 0,
    "ascendingIndex": 3291,
    "ifd": 0.9
  }
```
